arrow/arrow.py

```diff
@@ -131,7 +142,7 @@ def __init__(self, stoichiometry, rates, random_seed=0):
 
     def evolve(self, duration, state):
         steps, time, events, outcome = self.obsidian.evolve(duration, state)
-        occurrences = np.zeros(len(self.rates))
+        occurrences = np.zeros(self.stoichiometry.shape[0])
```

Comment:
I think `self.stoichiometry.shape[0]` is used in a couple places and should probably be factored out for convenience/consistency.